File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/ResponseHeaders.kt
package io.envoyproxy.envoymobile

/*
 * Headers representing an inbound response.
 */
class ResponseHeaders: Headers {
  /**
   * Internal constructor used by builders.
   *
   * @param headers: Headers to set.
   */
  internal constructor(headers: Map<String, List<String>>) : super(headers)

  /**
   * HTTP status code received with the response.
   */
  val httpStatus: Int? by lazy { value(":status")?.first()?.toIntOrNull() }
}

File: mobile/library/swift/ResponseHeaders.swift
import Foundation

/// Headers representing an inbound response.
@objcMembers
public final class ResponseHeaders: Headers {
  /// HTTP status code received with the response.
  public private(set) lazy var httpStatus: UInt? =
    self.value(forName: ":status")?.first.flatMap(UInt.init)

  /// Convert the headers back to a builder for mutation.
  ///
  /// - returns: The new builder.
  public func toResponseHeadersBuilder() -> ResponseHeadersBuilder {
    return ResponseHeadersBuilder(container: self.container)
  }
}
